uhyve-interface/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+//! # Uhyve Hypervisor Interface
+//!
+//! The uhyve hypercall interface works as follows:
+//!
+//! - On x86_64 you use a out port instruction. The address of the out_port corresponds to the
+//! hypercall you want to use. You can obtain it from the [`IoPorts`] enum. The data send to
+//! that port is the physical memory (in the VM) of the parameters of that hypercall.
+//! - On aarch64 you write to the respective [`HypercallAddress`]. The 64-bit data written to that location is the guest's physical memory address of the hypercall's parameter.
+
+#![no_std]
+
+// TODO: Throw this out, once https://github.com/rust-lang/rfcs/issues/2783 is resolved
+use num_enum::TryFromPrimitive;
+
+pub mod elf;
```

Comment:
Better keep this private. 🤫

Reply:
No. Why should we?

Reply:
Why should this be part of the public API? Everything besides the macro is an implementation detail.

Reply:
Because this is part of how uhyve will detect the kernel version. So, we might use this in uhyve or some other crates. This is truly public API from my perspective.

Reply:
As I said, I agree that the macro is part of the public API, but the rest will never be accessed via the Rust API from other crates or will it?